src/v/raft/consensus.cc

```diff
@@ -2250,6 +2250,30 @@ consensus::do_append_entries(append_entries_request&& r) {
     }
 }
 
+void consensus::validate_offset_translator_delta(
+  const protocol_metadata& meta, const storage::offset_stats& lstats) {
+    // do not validate if prev_log_delta is not set
+    if (meta.prev_log_delta < model::offset_delta{0}) {
+        return;
+    }
+    /**
+     * If request contain valid information and it is about to be appended
+     * to the log validate the offset translator delta consistency.
+     */
+    const auto last_delta = get_offset_delta(lstats, meta.prev_log_index);
+    if (
+      last_delta >= model::offset_delta{0}
+      && last_delta != meta.prev_log_delta) {
```

Comment:
why is information from the leader (meta.prev_log_index) being used on the follower to construct the value that is validated against the state arriving from the leader? that seems like a recipe for bias / false negatives?